engine/access/rpc/backend/backend_block_details.go

```diff
@@ -49,50 +58,50 @@ func (b *backendBlockDetails) GetLatestBlock(_ context.Context, isSealed bool) (
 		return nil, flow.BlockStatusUnknown, status.Errorf(codes.Internal, "could not get latest block: %v", err)
 	}
 
-	stat, err := b.getBlockStatus(block)
+	stat, err := b.getBlockStatus(ctx, block)
 	if err != nil {
 		return nil, stat, err
 	}
 	return block, stat, nil
 }
 
-func (b *backendBlockDetails) GetBlockByID(_ context.Context, id flow.Identifier) (*flow.Block, flow.BlockStatus, error) {
+func (b *backendBlockDetails) GetBlockByID(ctx context.Context, id flow.Identifier) (*flow.Block, flow.BlockStatus, error) {
 	block, err := b.blocks.ByID(id)
 	if err != nil {
 		return nil, flow.BlockStatusUnknown, rpc.ConvertStorageError(err)
 	}
 
-	stat, err := b.getBlockStatus(block)
+	stat, err := b.getBlockStatus(ctx, block)
 	if err != nil {
 		return nil, stat, err
 	}
 	return block, stat, nil
 }
 
-func (b *backendBlockDetails) GetBlockByHeight(_ context.Context, height uint64) (*flow.Block, flow.BlockStatus, error) {
+func (b *backendBlockDetails) GetBlockByHeight(ctx context.Context, height uint64) (*flow.Block, flow.BlockStatus, error) {
 	block, err := b.blocks.ByHeight(height)
 	if err != nil {
 		return nil, flow.BlockStatusUnknown, rpc.ConvertStorageError(err)
 	}
 
-	stat, err := b.getBlockStatus(block)
+	stat, err := b.getBlockStatus(ctx, block)
 	if err != nil {
 		return nil, stat, err
 	}
 	return block, stat, nil
 }
 
-func (b *backendBlockDetails) getBlockStatus(block *flow.Block) (flow.BlockStatus, error) {
+func (b *backendBlockDetails) getBlockStatus(ctx context.Context, block *flow.Block) (flow.BlockStatus, error) {
 	sealed, err := b.state.Sealed().Head()
 	if err != nil {
 		// In the RPC engine, if we encounter an error from the protocol state indicating state corruption,
 		// we should halt processing requests, but do throw an exception which might cause a crash:
 		// - It is unsafe to process requests if we have an internally bad state.
-		//   TODO: https://github.com/onflow/flow-go/issues/4028
 		// - We would like to avoid throwing an exception as a result of an Access API request by policy
 		//   because this can cause DOS potential
 		// - Since the protocol state is widely shared, we assume that in practice another component will
 		//   observe the protocol state error and throw an exception.
+		irrecoverable.Throw(ctx, fmt.Errorf("failed to lookup sealed header: %w", err))
```

Comment:
```suggestion
		irrecoverable.Throw(ctx, irrecoverable.NewExceptionf("failed to lookup sealed header: %w", err))
```

Reply:
strictly speaking `Throw` should terminate a function but golang doesn't know about this so we need to return some error, I think we have to return the same error that we throw.